includes/Connector.php

```diff
@@ -286,13 +288,23 @@ public function curl($url, $params_data = array(), $verb = "", $custom_method =
     /**
      * Throw the request exception
      *
-     * @param $message
+     * @param string $message
+     * @param resource $request
+     * @param string $data
      *
-     * @throws \RequestException
+     * @throws RequestException
      */
-    protected function throwRequestException($message)
+    protected function throwRequestException($message, $request = null, $data = null)
     {
         $requestException = new RequestException;
+
+        if ($request) {
+            $requestException->setContext(array(
+                "request_url"  => curl_getinfo($request, CURLINFO_EFFECTIVE_URL),
+                "request_body" => json_encode($data)
```

Comment:
@bartboy011 Is there any change that this `$data` could already be json_encoded and we're gonna double encode it?